src/ledger/LedgerManagerImpl.cpp

```diff
@@ -920,12 +926,30 @@ LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
     // step 2
     ltx.commit();
 
-    // step 3
-    hm.publishQueuedHistory();
-    hm.logAndUpdatePublishStatus();
+    if (!delayPublishToArchive)
+    {
+        // step 3
+        hm.publishQueuedHistory();
+        hm.logAndUpdatePublishStatus();
 
-    // step 4
-    mApp.getBucketManager().forgetUnreferencedBuckets();
+        // step 4
+        mApp.getBucketManager().forgetUnreferencedBuckets();
+    }
+    else
+    {
+        mDelayPublishToArchiveTimer.expires_from_now(
```

Comment:
I would suggest moving the delay logic to the actual publishing task. This codepath is quite sensitive (the node is atomically updating state of the ledger), so I recommend keeping the sequence of events as-is.

 Here we're snapshotting the current state of the ledger, then scheduling the publish work asynchronously https://github.com/stellar/stellar-core/blob/c2599d62274931c11ea7f36c5ee9255a202f5739/src/history/HistoryManagerImpl.cpp#L287. So I would suggest tweaking PublishWork to begin "work" with a small delay. Note that we're using a sequence of "works", so what you could do is add a WorkWithCallback at the front of the sequence, which succeeds if enough time has passed. Let me know if you have any questions about the Work subsystem, I think using it is the cleanest approach.